src/Service.Tests/Unittests/ConfigValidationUnitTests.cs

```diff
@@ -333,6 +333,110 @@ public void TestAddingRelationshipWithDisabledGraphQL()
             Assert.AreEqual(HttpStatusCode.ServiceUnavailable, ex.StatusCode);
         }
 
+        /// <summary>
+        /// Test to validate that we throw a validation error when GraphQL is enabled on the source entity and
+        /// the user defines multiple relationships between the same source and target entities.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(true, DisplayName = "Validate that exception is thrown when GQL is enabled and user defines multiple relationship between source and target entities.")]
+        [DataRow(false, DisplayName = "Validate that no exception is thrown when GQL is disabled and user defines multiple relationship between source and target entities.")]
```

Comment:
Something to consider: Should we log a warning (instead of throwing an exception which halts engine start-up) when graphQL is disabled for the entity?

Reply:
This was agreed upon in the scrum (including Jerry), that we should throw an exception.

Reply:
Right. No change in that.

My suggestion was that we could consider throwing a warning (instead of ignoring it) when the graphQL is disabled. Throwing a warning lets users know that something is wrong in the way they have defined the config, but, at the same time, engine does not crash since graphQL is disabled.

Reply:
But if the user does not want graphql, why should they see an unnecessary warning which should not bother them ideally?

Reply:
My initial thoughts were this ---> Because defining multiple relationships is an incorrect thing to have in the config file even if graphQL is disabled. Throwing a warning will let them know that the config is an invalid state.  It will be helpful if they have intentions of enabling graphQL in the future.


I guess, when they enable graphQL, they would hit this error and can modify the relationships. So, I don't have a strong preference as to throw/not throw a warning. 
 

Marking this resolved.